src/compiler/checker.ts

```diff
@@ -6193,11 +6198,9 @@ namespace ts {
                         emptyArray;
                     const publicProperties = flatMap<Symbol, ClassElement>(publicSymbolProps, p => serializePropertySymbolForClass(p, /*isStatic*/ false, baseTypes[0]));
                     // Consider static members empty if symbol also has function or module meaning - function namespacey emit will handle statics
-                    const staticMembers = symbol.flags & (SymbolFlags.Function | SymbolFlags.ValueModule)
-                        ? []
-                        : flatMap(filter(
+                    const staticMembers = flatMap(filter(
                             getPropertiesOfType(staticType),
-                            p => !(p.flags & SymbolFlags.Prototype) && p.escapedName !== "prototype"
+                            p => !(p.flags & SymbolFlags.Prototype) && p.escapedName !== "prototype" && p.valueDeclaration && isClassLike(p.valueDeclaration.parent)

```

Comment:
Should we extract this predicate to a function, so it's obvious each place is using the same predicate (albeit inverted)?

Reply:
yes, although it reveals that this usage is not quite the same as the others, because it needs to (1) accept `static` members (the opposite of the others) but (2) exclude `prototype` (same as the others)